src/app/zeko/da_layer/cli.ml

```diff
@@ -0,0 +1,37 @@
+open Core
+open Async
+
+let run_node =
+  ( "run-node"
+  , Command.async ~summary:"Run da layer node"
+      (let%map_open.Command db_dir =
+         flag "--db-dir"
+           (optional_with_default "da_db" string)
+           ~doc:"string Directory to store the database"
+       and port =
+         flag "--port"
+           (optional_with_default 8080 int)
+           ~doc:"int Port to listen on"
+       and node_to_sync =
+         flag "--node-to-sync" (optional string) ~doc:"string Node to sync with"
+       in
+       fun () ->
+         let signer = Sys.getenv_exn "MINA_PRIVATE_KEY" in
```

Comment:
This is quite suboptimal FWIW. The private key should not be in the memory of the DA node.
It should make a request to some external service to do the signing.
The external service could then either do it in hardware (e.g. FIDO2, TPM, hardware wallet),
or do it _properly_, that is, making sure that even if the private key is in memory, that it's never copied around, that it's not swapped to disk, etc. There are already many implementations of things in this direction.

Of course, it's complicated by the fact that we use Schnorr on Pallas,
but it should be a minor complication.
At the very least it should be a separate process.

You have to consider the attack vector where someone uses this API
to hack the DA node and get hold of their private key (or use it at least).

Reply:
Considering our constraints,
we should use the TPM functionality that the kernel provides.

This however constrains us to non-native curves, so we need to verify non-native signatures.
This is probably worth it however.